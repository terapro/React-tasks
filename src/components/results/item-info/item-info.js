import React from 'react';

import PropTypes from 'prop-types';

const ItemInfo = ({filmTitle, releaseDate, genre}) => {

  let title = filmTitle.length > 16 ? filmTitle.slice(0,18) + '...' :  filmTitle;

return (
  <div className='item-info'>
    <div className='item-info-details'>
      <div className='film-title'>
        {title}
      </div>
      <div className='release-date'>
        {releaseDate.slice(0,4)}
      </div>
    </div>
    <div className='item-info-genre'>
      {genre.join(', ')}
    </div>
  </div>
)
};

ItemInfo.propTypes = {
  filmTitle: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  genre: PropTypes.arrayOf(PropTypes.string).isRequired
};

export {ItemInfo};