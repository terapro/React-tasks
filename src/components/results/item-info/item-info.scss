.item-info {
  width: 200px;
  margin-top: 15px;
}
.item-info-details {
  display: flex;
  justify-content: space-between;
}

.film-title {
  font-size: 13px;
  text-transform: uppercase;
  font-weight: bold;
  text-overflow: clip;
  height: 12px;
  cursor: pointer;
}
.release-date {
  border: 1px solid #CCCCCC;
  font-size: 11px;
  padding: 2px 4px;
}

.item-info-genre {
  font-size: 10px;
}