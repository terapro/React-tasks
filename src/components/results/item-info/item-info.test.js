import React from 'react';
import {ItemInfo} from 'src/components/results/item-info/item-info.js';
import {shallow} from 'enzyme';

const mockFilmTitleText = 'Star Wars';
const mockFilmReleaseDateText = '2017-08-08';
const mockFilmGenreArray = ['comedy', 'action', 'drama'];

describe('<ItemInfo />', () => {
  describe('Rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(
        <ItemInfo filmTitle={mockFilmTitleText}
                  releaseDate={mockFilmReleaseDateText}
                  genre = {mockFilmGenreArray}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
