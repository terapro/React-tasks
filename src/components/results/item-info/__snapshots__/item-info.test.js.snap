// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemInfo /> Rendering should render the component 1`] = `
<div
  className="item-info"
>
  <div
    className="item-info-details"
  >
    <div
      className="film-title"
    >
      Star Wars
    </div>
    <div
      className="release-date"
    >
      2017
    </div>
  </div>
  <div
    className="item-info-genre"
  >
    comedy, action, drama
  </div>
</div>
`;
